internal/cloudtools/installation.go

```diff
@@ -22,9 +26,24 @@ func GetInstallationIDFromOwnerID(client *cloud.Client, serverURL, ownerID strin
 	if len(installations) == 0 {
 		return nil, nil
 	}
-	if len(installations) == 1 {
-		return installations[0], nil
+
+	// If there are more than 1 installations, return the first one that isn't deleted
+	// This allows support for creating new installations while old ones are sitting in deletion pending.
+	for _, installation := range installations {
+		if isNotDeletedState(installation.State) {
+			return installation, nil
+		}
```

Comment:
I had to modify this method in my local changes because it's not really doing what I expected. As it is, if there's already one (or more) installation in a delete state belonging to the same owner (PR), it will error rather than return `nil, nil` as I'd expect, which would then allow for a new creation to go through.

Reply:
Ive adjusted the final return to just be `nil, nil`

If there are no installations, a new one can be created (ie nil, nil)
If there is more than 0 installations, we find the first non-deletion-state one, and if it exists, return it
otherwise, all installations are in some deletion state of some kind, so we can create new ones (ie nil, nil)

I think technically with this change it's possible that some installations get stuck in deletion failed, but I personally prefer to have it that this wouldn't stop a user from creating new ones, and we can clean them up later.

Reply:
Thanks!